### DISCUSSION TITLE
How can i combine quarto with a Django project

### DISCUSSION THREAD

**Discussion by abdelkareemkobo** [quarto-dev/quarto-cli#6629] [Q&A] [answered]
### Description

I want to combine quarto auto rendering for .qmd files into a django website that i want to extend with other django power and features with python..etc but what i miss is how to use quarto withing django! is there any methods or third party packages that i can include to do this feature rendering .qmd file into a django pages! 
- why tell me why not just use quarto to build your website !
i am learning django and want to do multiple things for my website a lot of things and part of them is to create a blog for my two languages and other features that i know how to do with django and want to improve my django knowledge by using this and i don't know much javascript..etc and i am n't comfortable with quarto way to create the blogs within folders and each folder contain the post in this way it's hard to maintain for me!

**Comment by abdelkareemkobo** [comment]
another person interested in this https://www.reddit.com/r/webdev/comments/11fb5k1/anyone_has_used_quarto_and_django_for_web/

**Comment by mcanouil** [comment]
You can't or at least not in an easy way.
Django is a web framework which has nothing in common with how Quarto creates HTML content.
You, unfortunately have to pick a tool or find yourself a way to make the two coexist somehow.
Note that this problem is not specific to Quarto, i.e., Hugo, Jekyll, Wordpress, etc. You don't make them coexist you migrate from one to another based on the features you need.

**Reply by abdelkareemkobo** [reply]
If i want create a book or scientific documents i will definitely use quarto.
Bud for my blog need i just want the functionally to show python plots in the my html Django is thing can only be used with quarto! Quarto is an open source built on top of pandoc document is it difficult to integrate them with Django as third part app! Or any other way!
I really need this rendering power in Django can you guide me what should I build to get this even in the hard way or is there any other solutions!

**Reply by mcanouil** [reply, marked as answer]
You should look for Django and Jupyter Notebook integration, but these are out of Quarto scope.

**Reply by cderv** [reply]
If Django can deal with external HTML content that you can insert somewhere in the Django framework, then you could consider rendering your Quarto documents independently of your django project, and stick the HTML where needed. 

You would need to deal with resources though if not using `embed-resources: true` and if you want Django styling, then this become harder. You could deactivate everything with `minimal: true` and `theme: none`, build a custom format as an extension to have simpler HTML to embed. But you would loose some Quarto feature probably. 

If you consider the two independant though (not mixing theme for example), the scope would be how to serve static web page from within a Django framework. 

Otherwise, this would may be approach the other way: What would it take for django to call Quarto on some specific content when building a django website ? What should the output a quarto rendering look like ? Probably require a whole custom formats for compatible HTML. 

Anyhow, just ideas that goes to mind. 

Hope it helps

**Comment by joelsewhere** [comment]
Hi folks. I went through the process of setting up a template for a project that authors webpages via quarto and serves them with a django backend. This is only a day or two worth of work so I'm sure there are parts that can be improved, but it seems to be working pretty reliably at this point. Hope it helps. 

https://github.com/joelsewhere/quarto_django
